AVIATION ACCIDENTS FROM 1919 TO 2023

Intoduction: 

to identify the cause of accidents using this dataset that records the aviation accients from the year 1919 to the year 2023 and it will enable use to find a solution to prevent this in the near futer.

Objectives:

we will be able to identify ;
the cause of the accidents.
the place with most amount of accidents.
the fatality with the large amount of accidents.
the number of survivals and nonsurvivals in each category of the accidents.


Data Cleaning

In [27]:
#importing pandas
import pandas as pd
# importing the dataset
df_accidents=pd.read_csv("./aviation-accident-data-2023-05-16.csv")
# use .head() inorder to get access to the 1st few rows of the dataset
df_accidents.head()

date             type registration       operator fatalities  \
0  date unk.   Antonov An-12B       T-1206  Indonesian AF        NaN   
1  date unk.   Antonov An-12B       T-1204  Indonesian AF        NaN   
2  date unk.   Antonov An-12B       T-1201  Indonesian AF        NaN   
3  date unk.  Antonov An-12BK          NaN      Soviet AF        NaN   
4  date unk.  Antonov An-12BP   CCCP-11815      Soviet AF          0   

              location          country cat     year  
0                  NaN  Unknown country  U1  unknown  
1                  NaN  Unknown country  U1  unknown  
2                  NaN  Unknown country  U1  unknown  
3  Tiksi Airport (IKS)           Russia  A1  unknown  
4  Massawa Airport ...          Eritrea  A1  unknown

In [28]:
#using .info() to get a summary of the dataframe and also enable us to identify the columns with missing data
df_accidents.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23967 entries, 0 to 23966
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   date          23967 non-null  object
 1   type          23967 non-null  object
 2   registration  22419 non-null  object
 3   operator      23963 non-null  object
 4   fatalities    20029 non-null  object
 5   location      23019 non-null  object
 6   country       23967 non-null  object
 7   cat           23967 non-null  object
 8   year          23967 non-null  object
dtypes: object(9)
memory usage: 1.6+ MB


in the above we identify that the dataset has 9columns,23967 rows,23966 indexs and the columns contain strings. 
To iddentfy missing data 
the columns registration,operator,fatalities and location all have missing data since their non-null values don't add up to the total number of colums which is 23967.


In [29]:
#ditecting NaNs
df_accidents.isna().sum()

date               0
type               0
registration    1548
operator           4
fatalities      3938
location         948
country            0
cat                0
year               0
dtype: int64

In [30]:
#filling the missing values using .fillna() 
missing_cols=['registration','operator','fatalities','location']
df_accidents[missing_cols]=df_accidents[missing_cols].fillna('Unknown')
df_accidents.isna().sum()

date            0
type            0
registration    0
operator        0
fatalities      0
location        0
country         0
cat             0
year            0
dtype: int64

In [31]:
#to find if there are any duplicates and how many they are in the dataset
#true means duplicate and false means not a duplicate
df_accidents.duplicated().value_counts()

False    23852
True       115
dtype: int64

In [32]:
#using .drop_duplicates() to remove duplicates
#after using .drop_duplicates() we can see that there are no more duplicates in the dataset
df_accidents=df_accidents.drop_duplicates()
df_accidents.duplicated().value_counts()

False    23852
dtype: int64

In [33]:
# to save the cleaned dataset to a new csv file
df_accidents.to_csv("cleaned_aviation_accident_data.csv", index=False)